### PR TITLE
DateIntervalType: 'invert' should not inherit the 'required' option

### PR DESCRIPTION
**Pull request by galeaspablo**
| Q             | A
| ------------- | ---
| Branch?       | 3.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #20876
| License       | MIT
| Doc PR        | -

As explained in #20876, 

> In the DateIntervalType, there is a field, called 'invert', that allows for negative intervals. This is outputted as a checkbox. Which is fine, but it shouldn't be required.

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php [hunk 1]
```diff
@@ -134,6 +134,8 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     $builder->add($childForm);
                 }
             }
+            // Invert should not inherit the required option
+            $invertOptions['required'] = false;
```
Comment (ogizanagi) [inline src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php]: What about 
```diff
diff --git a/src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php b/src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php
diff --git a/src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php b/src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php
index fb2b7d7..e4f46d8 100644
--- a/src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php
+++ b/src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php
@@ -107,9 +107,6 @@ class DateIntervalType extends AbstractType
                     }
                 }
             }
-            $invertOptions = array(
-                'error_bubbling' => true,
-            );
             // Append generic carry-along options
             foreach (array('required', 'translation_domain') as $passOpt) {
                 foreach ($this->timeParts as $part) {
@@ -117,10 +114,8 @@ class DateIntervalType extends AbstractType
                         $childOptions[$part][$passOpt] = $options[$passOpt];
                     }
                 }
-                if ($options['with_invert']) {
-                    $invertOptions[$passOpt] = $options[$passOpt];
-                }
             }
+
             foreach ($this->timeParts as $part) {
                 if ($options['with_'.$part]) {
                     $childForm = $builder->create($part, self::$widgets[$options['widget']], $childOptions[$part]);
@@ -135,7 +130,11 @@ class DateIntervalType extends AbstractType
                 }
             }
             if ($options['with_invert']) {
-                $builder->add('invert', 'Symfony\Component\Form\Extension\Core\Type\CheckboxType', $invertOptions);
+                $builder->add('invert', 'Symfony\Component\Form\Extension\Core\Type\CheckboxType', array(
+                    'error_bubbling' => true,
+                    'required' => false,
+                    'translation_domain' => $options['translation_domain'],
+                ));
             }
             $builder->addViewTransformer(new DateIntervalToArrayTransformer($parts, 'text' === $options['widget']));
         }
```

instead? Not much different, but simplify things a little bit.

Comment (galeaspablo) [inline src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php]: Looks good to me. Changed. :+1:

File: src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php [hunk 2]
```diff
@@ -107,20 +107,15 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
                 foreach ($this->timeParts as $part) {
                     if ($options['with_'.$part]) {
                         $childOptions[$part][$passOpt] = $options[$passOpt];
                     }
                 }
-                if ($options['with_invert']) {
-                    $invertOptions[$passOpt] = $options[$passOpt];
-                }
             }
+
```
Comment (ogizanagi) [inline src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php]: Should be reverted

**Comment by ogizanagi** [comment]
👍 

Status: Reviewed

File: src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php [hunk 3]
```diff
@@ -135,7 +129,11 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 }
             }
             if ($options['with_invert']) {
-                $builder->add('invert', 'Symfony\Component\Form\Extension\Core\Type\CheckboxType', $invertOptions);
+                $builder->add('invert', 'Symfony\Component\Form\Extension\Core\Type\CheckboxType', array(
+                    'error_bubbling' => true,
+                    'required' => false,
+                    'translation_domain' => $options['translation_domain'],
```
Comment (xabbuh) [inline src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php]: Is the `translation_domain` option needed here at all?

Comment (ogizanagi) [inline src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php]: Anyway it should be kept IMHO. It's used to translate the `Invert` label, so it makes sense to use the same translation domain as for the whole widget.

**Comment by fabpot** [comment]
Thank you @galeaspablo.

File: src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php (hunk 3 above)
Comment (xabbuh) [inline src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php]: @ogizanagi Indeed, you are right.

**Pull request closed without merging**